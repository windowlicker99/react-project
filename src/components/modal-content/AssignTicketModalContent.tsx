import React, { ChangeEventHandler, FC } from 'react';
import { FormikValues } from 'formik';
import { Textarea } from '@components/common/form-controls/Textarea';
import { SelectComponent } from '@components/common/form-controls/SelectComponent';
import { IOption } from '@interfaces/interfaces';

interface IAssignTicketModalContent {
  values: FormikValues;
  options: IOption[];
  onChange: ChangeEventHandler;
  onSelectChange: (name: string, value: IOption) => void;
}

export const AssignTicketModalContent: FC<IAssignTicketModalContent> = ({
  values,
  options,
  onChange,
  onSelectChange,
}) => (
  <div className="modal-info">
    <SelectComponent
      name="user"
      value={values.user}
      options={options}
      label="Assign to"
      placeholder=""
      onChange={onSelectChange}
    />
    <Textarea placeholder="Add a message (optional)" name="message" value={values.message} onChange={onChange} />
  </div>
);
