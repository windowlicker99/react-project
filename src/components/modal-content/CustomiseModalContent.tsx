import React, { ChangeEventHandler, FC, FormEventHandler } from 'react';
import { FormikValues } from 'formik';
import { Checkbox } from '@components/common/form-controls/Checkbox';

interface IColumnField {
  label: string;
  name: string;
}

interface ICustomiseModalContent {
  handleSubmit: FormEventHandler<HTMLFormElement>;
  handleChange: ChangeEventHandler<HTMLInputElement>;
  values: FormikValues;
  columns: IColumnField[];
}

export const CustomiseModalContent: FC<ICustomiseModalContent> = ({ values, columns, handleChange, handleSubmit }) => (
  <div className="modal-info">
    <form onSubmit={handleSubmit}>
      <ul className="customise-list">
        {columns.map(({ label, name }) => (
          <li className="customise-item" key={name}>
            <button type="button" className="customise-move" />
            <Checkbox name={name} checked={values[name]} onChange={handleChange} label={label} />
          </li>
        ))}
      </ul>
    </form>
  </div>
);
