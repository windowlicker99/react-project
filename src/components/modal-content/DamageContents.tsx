import { FormikValues, useFormikContext } from 'formik';
import React, { FC } from 'react';
import { SelectComponent } from '@components/common/form-controls/SelectComponent';
import { setDamagePotential } from '@/constants/formsElements';
import { deleteInitialValues } from '@/utils/deleteInitialValues';

const title = [{ name: 'Confidentiality' }, { name: 'Integrity' }, { name: 'Availability' }, { name: 'Authenticity' }];

const formInputs = deleteInitialValues(setDamagePotential);

export const DamageContent: FC = () => {
  const { values, setFieldValue } = useFormikContext<FormikValues>();
  return (
    <div className="damage-content">
      {title.map(({ name }) => (
        <React.Fragment key={name}>
          <span className="damage-title">{name}</span>
          <div className="damage-item">
            {Object.keys(formInputs).map((input) => (
              <SelectComponent
                key={input}
                onChange={setFieldValue}
                {...formInputs[input]}
                value={values[name][input]}
                name={`${name}.${input}`}
              />
            ))}
          </div>
        </React.Fragment>
      ))}
    </div>
  );
};
