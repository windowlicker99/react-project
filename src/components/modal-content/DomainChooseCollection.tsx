// markup only
import React, { FC } from 'react';
import { CardDropdown } from '@components/common/Ð¡ardDropdown';
import { ModalControl } from '@components/common/ModalControl';

export const DomainChooseContent: FC = () => (
  <div className="domains-aside-item">
    <ModalControl />
    <CardDropdown name="domain" />
  </div>
);
