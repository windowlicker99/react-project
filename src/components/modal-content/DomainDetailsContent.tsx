import React, { FC } from 'react';
import { FormikProps, FormikValues } from 'formik';
import { Input } from '@components/common/form-controls/Input';
import { BadgeRoundGroup } from '@components/common/BadgeRoundGroup';
import { SelectComponent } from '@components/common/form-controls/SelectComponent';
import { FormInputSwitch } from '@components/common/form-controls/FormInputSwitch';
import { IOption } from '@interfaces/interfaces';
import { domainCreationElements } from '@constants/formsElements';
import { deleteInitialValues } from '@utils/deleteInitialValues';

interface IDomainDetailsContent {
  handleSelectChange: (name: string, value: IOption) => void;
}

export const DomainDetailsContent: FC<IDomainDetailsContent & Partial<FormikProps<FormikValues>>> = ({
  values,
  errors,
  touched,
  handleBlur,
  handleChange,
  handleSelectChange,
}) => {
  const { name, usage, connectivity, communication, color } = deleteInitialValues(domainCreationElements);

  return (
    <div className="domains-aside-item">
      <div className="modal-box">
        <Input
          {...name}
          error={touched[name.name] && (errors[name.name] as string)}
          onChange={handleChange}
          value={values[name.name]}
          onBlur={handleBlur}
        />
        <Input
          {...usage}
          error={touched[usage.name] && (errors[usage.name] as string)}
          onChange={handleChange}
          value={values[usage.name]}
          onBlur={handleBlur}
        />
        <FormInputSwitch value={values[connectivity.name]} {...connectivity} onChange={handleChange} />
        <SelectComponent
          error={touched[communication.name] && (errors[communication.name] as string)}
          value={values[communication.name]}
          {...communication}
          onChange={handleSelectChange}
        />
      </div>
      <BadgeRoundGroup value={values[color.name]} {...color} onChange={handleChange} />
    </div>
  );
};
