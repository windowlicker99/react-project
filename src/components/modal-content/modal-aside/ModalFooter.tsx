import React, { FC } from 'react';
import { Btn } from '@components/common/form-controls/Btn';
import { Switch } from '@components/common/form-controls/Switch';

interface IModalFooterProps {
  applyBtn?: string;
  disabledButton?: boolean;
  onApply?: () => void;
  onBack?: () => void;
  showDuplicateSwitch?: boolean;
}

export const ModalFooter: FC<IModalFooterProps> = ({
  onBack,
  onApply,
  applyBtn,
  disabledButton = false,
  showDuplicateSwitch = false,
}) => (
  <div className="modal-footer-control">
    <div className="modal-footer-item">
      {onBack && (
        <Btn
          btnStatus="btn-link"
          disabledBtn={disabledButton}
          btnIcon="icon-arrow-left"
          btnText="Back"
          onClick={onBack}
          aria-hidden="true"
        />
      )}
      {showDuplicateSwitch && (
        <>
          <Switch />
          <span className="modal-footer-info">Duplicate all elements with selected (Element type, Element name)</span>
        </>
      )}
    </div>
    {applyBtn && <Btn btnStatus="btn-lg-primary btn-stretch" btnIcon="" btnText={applyBtn} onClick={onApply} />}
  </div>
);
