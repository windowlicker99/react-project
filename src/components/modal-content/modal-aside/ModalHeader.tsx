import React, { FC } from 'react';
import { PageTabs } from '@components/common/PageTabs';

interface IModalHeader {
  title?: string;
  subtitle?: string;
  activeTab?: string;
  tabsList?: string[];
  onBack?: () => void;
  onChooseTabClick?: (str: string) => void;
  onClose: () => void;
  step?: number;
  maxStep?: number;
}

export const ModalHeader: FC<IModalHeader> = ({
  title = '',
  subtitle,
  activeTab = '',
  tabsList,
  // eslint-disable-next-line @typescript-eslint/no-empty-function
  onChooseTabClick = () => {},
  onClose,
  step,
  maxStep = 2,
  onBack,
}) => (
  <>
    <div className="modal-heading">
      <div className="modal-header-content">
        {Boolean(step) && step <= maxStep && (
          <span className="modal-subtitle">
            Step {step}/{maxStep}
          </span>
        )}
        <span className="modal-title">{title}</span>
        <span className="modal-subtitle">{subtitle}</span>
        {Boolean(onBack) && (
          <span className="modal-subtitle">
            <span className="icon-arrow-left" aria-hidden="true" onClick={onBack} />
          </span>
        )}
      </div>
      <button type="button" className="btn btn-xs-secondary" onClick={onClose}>
        <span className="icon-close" />
      </button>
    </div>
    {Boolean(tabsList) && <PageTabs active={activeTab} tabs={tabsList} onClick={onChooseTabClick} />}
  </>
);
