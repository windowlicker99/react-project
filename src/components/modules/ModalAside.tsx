import React, { FC, ReactNode } from 'react';
import classNames from 'classnames';

interface IModalAsideProps {
  isOpen: boolean;
  header?: ReactNode;
  footer?: ReactNode;
}

export const ModalAside: FC<IModalAsideProps> = ({ isOpen, header, footer, children }) => (
  <div className={classNames('modal modal-aside', { show: isOpen })}>
    {/* <!-- add class show --> */}
    <div className="modal-content">
      {header && <div className="modal-header">{header}</div>}
      <div className="modal-body">{children}</div>
      {footer && <div className="modal-footer">{footer}</div>}
    </div>
  </div>
);
