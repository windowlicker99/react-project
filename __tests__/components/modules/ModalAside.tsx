import React from 'react';
import { render } from '@testing-library/react';
import { ModalAside } from '@components/modules/ModalAside';
import { DomainDetailsContent } from '@components/modal-content/DomainDetailsContent';
import { ModalHeader } from '@components/modal-content/modal-aside/ModalHeader';
import { ModalFooter } from '@components/modal-content/modal-aside/ModalFooter';

describe('ModalAside tests', () => {
  const mockErrors = { test: 'Required' };
  const mockTouched = { test: true };
  const mockValues = { test: 'Test1' };

  it('Should render', () => {
    const { container } = render(
      <ModalAside
        isOpen
        header={<ModalHeader title="Test" onClose={jest.fn()} />}
        footer={<ModalFooter applyBtn="Test" onApply={jest.fn()} />}
      >
        <DomainDetailsContent
          values={mockValues}
          touched={mockTouched}
          errors={mockErrors}
          handleBlur={jest.fn()}
          handleChange={jest.fn()}
          handleSelectChange={jest.fn()}
        />
      </ModalAside>
    );
    expect(container.getElementsByClassName('modal').length).toEqual(1);
  });

  it('Should match snapshot', () => {
    const { asFragment } = render(
      <ModalAside
        isOpen
        header={<ModalHeader title="Test" onClose={jest.fn()} />}
        footer={<ModalFooter applyBtn="Test" onApply={jest.fn()} />}
      >
        <DomainDetailsContent
          values={mockValues}
          touched={mockTouched}
          errors={mockErrors}
          handleBlur={jest.fn()}
          handleChange={jest.fn()}
          handleSelectChange={jest.fn()}
        />
      </ModalAside>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
