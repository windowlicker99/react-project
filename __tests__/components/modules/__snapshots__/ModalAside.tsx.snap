// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalAside tests Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="modal modal-aside show"
  >
    <div
      class="modal-content"
    >
      <div
        class="modal-header"
      >
        <div
          class="modal-heading"
        >
          <div
            class="modal-header-content"
          >
            <span
              class="modal-title"
            >
              Test
            </span>
            <span
              class="modal-subtitle"
            />
          </div>
          <button
            class="btn btn-xs-secondary"
            type="button"
          >
            <span
              class="icon-close"
            />
          </button>
        </div>
      </div>
      <div
        class="modal-body"
      >
        <div
          class="domains-aside-item"
        >
          <div
            class="modal-box"
          >
            <div
              class="form-group"
            >
              <label
                class="form-label"
                for="name"
              >
                Domain Name
              </label>
              <input
                class="form-control"
                maxlength="25"
                name="name"
                placeholder="Enter Domain Name"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                class="form-label"
                for="domainUsage"
              >
                Domain Usage
              </label>
              <input
                class="form-control"
                maxlength="25"
                name="domainUsage"
                placeholder="Enter Domain Usage"
                value=""
              />
            </div>
            <div
              class="form-input-switch"
            >
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="onlineConnectivity"
                >
                  Online Connectivity
                </label>
                <input
                  class="form-control"
                  disabled=""
                  name="onlineConnectivity"
                  placeholder="Online"
                />
              </div>
              <div
                class="switch"
              >
                <input
                  class="switch-input"
                  id="connectivity"
                  name="connectivity"
                  type="checkbox"
                />
                <label
                  class="switch-label"
                  for="connectivity"
                >
                  <span
                    class="switch-text"
                  />
                  <span
                    class="switch-trigger"
                  />
                </label>
              </div>
            </div>
            <div
              class="form-group"
            >
              <label
                class="form-label"
                for="exampleFormControlInput1"
              >
                Communication
              </label>
              <div
                class="select-component css-2b097c-container"
              >
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class="react-select__control css-1pgkwyd-control"
                >
                  <div
                    class="react-select__value-container react-select__value-container--is-multi css-g1d714-ValueContainer"
                  >
                    <div
                      class="react-select__placeholder css-oco7ot-placeholder"
                    >
                      Communication
                    </div>
                    <div
                      class="css-b8ldur-Input"
                    >
                      <div
                        class="react-select__input"
                        style="display: inline-block;"
                      >
                        <input
                          aria-autocomplete="list"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          id="react-select-3-input"
                          spellcheck="false"
                          style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                        <div
                          style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                  >
                    <div
                      aria-hidden="true"
                      class="react-select__indicator react-select__dropdown-indicator css-tlfecz-indicatorContainer"
                    >
                      <div
                        class="form-arrow icon-arrow-border-down"
                      />
                    </div>
                  </div>
                </div>
                <div>
                  <input
                    name="communication"
                    type="hidden"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="badge-round-group"
          >
            <span
              class="badge-round-title"
            >
              Domain Color
            </span>
            <div
              class="badge-round-content"
            >
              <div
                class="badge-round #ffd700"
                style="background-color: rgb(255, 215, 0);"
              >
                <input
                  class="badge-round-input"
                  id="color-#ffd700"
                  name="color"
                  type="radio"
                  value="#ffd700"
                />
                <label
                  class="badge-round-label"
                  for="color-#ffd700"
                />
              </div>
              <div
                class="badge-round #ffca6a"
                style="background-color: rgb(255, 202, 106);"
              >
                <input
                  class="badge-round-input"
                  id="color-#ffca6a"
                  name="color"
                  type="radio"
                  value="#ffca6a"
                />
                <label
                  class="badge-round-label"
                  for="color-#ffca6a"
                />
              </div>
              <div
                class="badge-round #ffa027"
                style="background-color: rgb(255, 160, 39);"
              >
                <input
                  class="badge-round-input"
                  id="color-#ffa027"
                  name="color"
                  type="radio"
                  value="#ffa027"
                />
                <label
                  class="badge-round-label"
                  for="color-#ffa027"
                />
              </div>
              <div
                class="badge-round #fb5e06"
                style="background-color: rgb(251, 94, 6);"
              >
                <input
                  class="badge-round-input"
                  id="color-#fb5e06"
                  name="color"
                  type="radio"
                  value="#fb5e06"
                />
                <label
                  class="badge-round-label"
                  for="color-#fb5e06"
                />
              </div>
              <div
                class="badge-round #f26563"
                style="background-color: rgb(242, 101, 99);"
              >
                <input
                  class="badge-round-input"
                  id="color-#f26563"
                  name="color"
                  type="radio"
                  value="#f26563"
                />
                <label
                  class="badge-round-label"
                  for="color-#f26563"
                />
              </div>
              <div
                class="badge-round #fc2033"
                style="background-color: rgb(252, 32, 51);"
              >
                <input
                  class="badge-round-input"
                  id="color-#fc2033"
                  name="color"
                  type="radio"
                  value="#fc2033"
                />
                <label
                  class="badge-round-label"
                  for="color-#fc2033"
                />
              </div>
              <div
                class="badge-round #c50029"
                style="background-color: rgb(197, 0, 41);"
              >
                <input
                  class="badge-round-input"
                  id="color-#c50029"
                  name="color"
                  type="radio"
                  value="#c50029"
                />
                <label
                  class="badge-round-label"
                  for="color-#c50029"
                />
              </div>
              <div
                class="badge-round #ff3a88"
                style="background-color: rgb(255, 58, 136);"
              >
                <input
                  class="badge-round-input"
                  id="color-#ff3a88"
                  name="color"
                  type="radio"
                  value="#ff3a88"
                />
                <label
                  class="badge-round-label"
                  for="color-#ff3a88"
                />
              </div>
              <div
                class="badge-round #AD6EFF"
                style="background-color: rgb(173, 110, 255);"
              >
                <input
                  class="badge-round-input"
                  id="color-#AD6EFF"
                  name="color"
                  type="radio"
                  value="#AD6EFF"
                />
                <label
                  class="badge-round-label"
                  for="color-#AD6EFF"
                />
              </div>
              <div
                class="badge-round #e47bf1"
                style="background-color: rgb(228, 123, 241);"
              >
                <input
                  class="badge-round-input"
                  id="color-#e47bf1"
                  name="color"
                  type="radio"
                  value="#e47bf1"
                />
                <label
                  class="badge-round-label"
                  for="color-#e47bf1"
                />
              </div>
              <div
                class="badge-round #8555c9"
                style="background-color: rgb(133, 85, 201);"
              >
                <input
                  class="badge-round-input"
                  id="color-#8555c9"
                  name="color"
                  type="radio"
                  value="#8555c9"
                />
                <label
                  class="badge-round-label"
                  for="color-#8555c9"
                />
              </div>
              <div
                class="badge-round #7674fd"
                style="background-color: rgb(118, 116, 253);"
              >
                <input
                  class="badge-round-input"
                  id="color-#7674fd"
                  name="color"
                  type="radio"
                  value="#7674fd"
                />
                <label
                  class="badge-round-label"
                  for="color-#7674fd"
                />
              </div>
              <div
                class="badge-round #4f98ff"
                style="background-color: rgb(79, 152, 255);"
              >
                <input
                  class="badge-round-input"
                  id="color-#4f98ff"
                  name="color"
                  type="radio"
                  value="#4f98ff"
                />
                <label
                  class="badge-round-label"
                  for="color-#4f98ff"
                />
              </div>
              <div
                class="badge-round #00c3ff"
                style="background-color: rgb(0, 195, 255);"
              >
                <input
                  class="badge-round-input"
                  id="color-#00c3ff"
                  name="color"
                  type="radio"
                  value="#00c3ff"
                />
                <label
                  class="badge-round-label"
                  for="color-#00c3ff"
                />
              </div>
              <div
                class="badge-round #bec8e7"
                style="background-color: rgb(190, 200, 231);"
              >
                <input
                  class="badge-round-input"
                  id="color-#bec8e7"
                  name="color"
                  type="radio"
                  value="#bec8e7"
                />
                <label
                  class="badge-round-label"
                  for="color-#bec8e7"
                />
              </div>
              <div
                class="badge-round #15eed4"
                style="background-color: rgb(21, 238, 212);"
              >
                <input
                  class="badge-round-input"
                  id="color-#15eed4"
                  name="color"
                  type="radio"
                  value="#15eed4"
                />
                <label
                  class="badge-round-label"
                  for="color-#15eed4"
                />
              </div>
              <div
                class="badge-round #10c9a1"
                style="background-color: rgb(16, 201, 161);"
              >
                <input
                  class="badge-round-input"
                  id="color-#10c9a1"
                  name="color"
                  type="radio"
                  value="#10c9a1"
                />
                <label
                  class="badge-round-label"
                  for="color-#10c9a1"
                />
              </div>
              <div
                class="badge-round #00ab9c"
                style="background-color: rgb(0, 171, 156);"
              >
                <input
                  class="badge-round-input"
                  id="color-#00ab9c"
                  name="color"
                  type="radio"
                  value="#00ab9c"
                />
                <label
                  class="badge-round-label"
                  for="color-#00ab9c"
                />
              </div>
              <div
                class="badge-round #cbdb15"
                style="background-color: rgb(203, 219, 21);"
              >
                <input
                  class="badge-round-input"
                  id="color-#cbdb15"
                  name="color"
                  type="radio"
                  value="#cbdb15"
                />
                <label
                  class="badge-round-label"
                  for="color-#cbdb15"
                />
              </div>
              <div
                class="badge-round #49c768"
                style="background-color: rgb(73, 199, 104);"
              >
                <input
                  class="badge-round-input"
                  id="color-#49c768"
                  name="color"
                  type="radio"
                  value="#49c768"
                />
                <label
                  class="badge-round-label"
                  for="color-#49c768"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="modal-footer"
      >
        <div
          class="modal-footer-control"
        >
          <div
            class="modal-footer-item"
          />
          <button
            class="btn btn-lg-primary btn-stretch"
            type="button"
          >
            <span
              class="btn-icon "
            />
            <span
              class="btn-text"
            >
              Test
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
